src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -1398,7 +1378,7 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
   #prepareNextBlock = (
     targetBlock: Block,
     parentBlock: Block,
-    transactionHash: Buffer
+    transactionIndex: number
```

Comment:
Instead of checking if we are done by comparing hashes let's just loop up to (and including) the index the transaction is at. No hash comparisons needed!

Reply:
done!